src/core/mesh/qgstriangularmesh.cpp

```diff
@@ -143,6 +146,15 @@ void QgsTriangularMesh::update( QgsMesh *nativeMesh, QgsRenderContext *context )
     ENP_centroid( poly, cx, cy );
     mNativeMeshFaceCentroids[i] = QgsMeshVertex( cx, cy );
   }
+
+  // CALCULATE SPATIAL INDEX
+  for ( int i = 0; i < mTriangularMesh.faces.size(); ++i )
```

Comment:
Not sure what kind of datatype mTrisangularMesh.faces is, but if it's a container, better use an iterator

Reply:
I also need `i` few lines lower to add geometry to spatial index

Reply:
If it's performance critical code, better iterate and use a separate counter. If not, no worries.